File: C++/virtual-world-simulator/src/Entity/Plants/Grass.cpp
#include "Grass.h"

Grass::Grass(World* world, Point position) :
	Plant { AssetManager::getAssetManager()->getAsset("grass"), world, position, 'G', 0, SpreadChance::MEDIUM }
{

}

Grass::Grass(World* world, Point position, int lifeSpan, int strength) :
	Plant { AssetManager::getAssetManager()->getAsset("grass"), world, position, 'G', strength, lifeSpan, SpreadChance::MEDIUM, }
{
}

void Grass::update(std::fstream& logFile)
{
	this->spread<Grass>(logFile);
	Entity::update(logFile);
}

File: Python/virtual-world-simulator/src/Entity/Plants/Grass.py
import pygame
from src.Entity.Plants.Plant import *


class Grass(Plant):

    def __init__(self, world, position):
        temp = pygame.image.load('assets/grass.png')
        scaled = pygame.transform.scale(temp, (world.scale, world.scale))
        super().__init__(scaled, world, position, 'G', 0, SpreadChance.MEDIUM)

    def update(self):
        spreadPosition = self.spread()
        if spreadPosition is not None:
            element = Grass(self._world, spreadPosition)
            self._world.setMapElement(spreadPosition[0], spreadPosition[1], element)
        super().update()